File: ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        int count1[30] = {0};
        int count2[30] = {0};
        for(int i=0;i<ransomNote.size();i++)
            count1[ransomNote[i] - 'a']++;
        for(int i=0;i<magazine.size();i++)
            count2[magazine[i] - 'a']++;
        for(int i=0;i<26;i++)
            if(count2[i] < count1[i])
                return false;
        return true;
    }
};

File: May LeetCoding Challenge/Ransom Note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        if (ransomNote.length() > magazine.length()) return false;
        int[] count = new int[30];
        for(char c:magazine.toCharArray())count[c-'a']++;
        for(char c:ransomNote.toCharArray())
        {
            count[c-'a']--;
            if(count[c-'a'] == -1)return false;
        }
        return true;
    }
}
